device/coordinate_manager.h

```diff
@@ -6,74 +6,161 @@
+        const tt_xy_pair& dram_grid_size,
+        const std::vector<tt_xy_pair>& dram_cores,
+        const size_t dram_harvesting_mask,
+        const tt_xy_pair& eth_grid_size,
+        const std::vector<tt_xy_pair>& eth_cores,
+        const tt_xy_pair& arc_grid_size,
+        const std::vector<tt_xy_pair>& arc_cores,
+        const tt_xy_pair& pcie_grid_size,
+        const std::vector<tt_xy_pair>& pcie_cores) :
+        tensix_grid_size(tensix_grid_size),
+        tensix_cores(tensix_cores),
+        tensix_harvesting_mask(tensix_harvesting_mask),
+        dram_grid_size(dram_grid_size),
+        dram_cores(dram_cores),
+        dram_harvesting_mask(dram_harvesting_mask),
+        eth_grid_size(eth_grid_size),
+        eth_cores(eth_cores),
+        arc_grid_size(arc_grid_size),
+        arc_cores(arc_cores),
+        pcie_grid_size(pcie_grid_size),
+        pcie_cores(pcie_cores) {}
+
+    virtual void tensix_harvesting(const size_t harvesting_mask);
+
+    virtual void dram_harvesting(const size_t dram_harvesting_mask);
+
+    virtual void translate_eth_coords();
+
+    virtual void translate_arc_coords();
+
+    virtual void translate_pcie_coords();
 
-    virtual void perform_harvesting(std::size_t harvesting_mask);
+    static std::unique_ptr<CoordinateManager> get_coordinate_manager(
+        tt::ARCH arch,
+        const tt_xy_pair& tensix_grid_size,
+        const std::vector<tt_xy_pair>& tensix_cores,
+        const size_t tensix_harvesting_mask,
+        const tt_xy_pair& dram_grid_size,
+        const std::vector<tt_xy_pair>& dram_cores,
+        const size_t dram_harvesting_mask,
+        const tt_xy_pair& eth_grid_size,
+        const std::vector<tt_xy_pair>& eth_cores,
+        const tt_xy_pair& arc_grid_size,
+        const std::vector<tt_xy_pair>& arc_cores,
+        const tt_xy_pair& pcie_grid_size,
+        const std::vector<tt_xy_pair>& pcie_cores);
 
-    virtual tt_physical_coords to_physical_coords(tt_logical_coords logical_coords);
-    virtual tt_translated_coords to_translated_coords(tt_logical_coords logical_coords);
-    virtual tt_virtual_coords to_virtual_coords(tt_logical_coords logical_coords);
+    static std::unique_ptr<CoordinateManager> get_coordinate_manager(
+        tt::ARCH arch, const size_t tensix_harvesting_mask = 0, const size_t dram_harvesting_mask = 0);
 
-    virtual tt_logical_coords to_logical_coords(tt_physical_coords physical_coords);
-    virtual tt_virtual_coords to_virtual_coords(tt_physical_coords physical_coords);
-    virtual tt_translated_coords to_translated_coords(tt_physical_coords physical_coords);
+    CoordinateManager(CoordinateManager& other) = default;
 
-    virtual tt_logical_coords to_logical_coords(tt_virtual_coords virtual_coords);
-    virtual tt_physical_coords to_physical_coords(tt_virtual_coords virtual_coords);
-    virtual tt_translated_coords to_translated_coords(tt_virtual_coords virtual_coords);
+    CoreCoord to_physical(const CoreCoord core_coord);
```

Comment:
Don't understand the question really. If you think what if the coordinate specified is not valid for given coordinate system, there will be an exception there

Reply:
User of tt-lens says, read 128 bytes from noc0(1,1). My question is, do I need to check which core is on noc0(1,1) and provide that information to UMD so I don't get exception or that information won't be checked for coordinates that don't require `core_type`?

Reply:
Yes, makes sense, I have code for that which is integrating core coords in Cluster, but it is not ready for PR yet. That is not possible through Coordinate Manager at the moment, meaning you need to specify coordinate type, but nice catch. I t. Again, it won't be in this PR, it will be in one of the following

Reply:
#348